core/controllers/reader.py

```diff
@@ -56,6 +56,38 @@
 
 def _get_exploration_player_data(
         exploration_id, version, collection_id, can_edit):
+    """Returns a dict of exploration player data.
+
+    Args:
+        exploration_id: str. The ID of the exploration.
+        version: int or None. The version of the exploration.
+        collection_id: str. ID of the collection.
+        can_edit: bool. Whether the given user can edit this activity.
+
+    Returns:
+        dict. A dict of exploration player data.
+        The keys and values of the dict are as follows:
+        - 'INTERACTION_SPECS': dict. A dict containing the full specs of each
+            interaction. Contains interaction ID and a list of instances of
+            all interactions.
+        - 'DEFAULT_TWITTER_SHARE_MESSAGE_PLAYER': str. Text for the Twitter
+            share message.
+        - 'additional_angular_modules': list. A de-duplicated list of strings,
+            each representing an additional angular module that should be
+            loaded.
+        - 'can_edit': bool. Whether the given user can edit this activity.
+        - 'dependencies_html': str. The additional HTML to insert on the page.
+        - 'exploration_title': str. Title of exploration.
+        - 'exploration_version': int. The version of the exploration.
+        - 'collection_id': str. ID of the collection.
+        - 'collection_title': str. Title of collection.
+        - 'interaction_templates': str. The HTML bodies for the given list of
```

Comment:
I don't see a "given" list of interaction IDs in the args for this function...?

Reply:
Yes, interaction IDs is computed by get_interaction_ids().

Reply:
The HTML bodies of the interactions required by this exploration.

(something like that, perhaps?)

Reply:
The HTML bodies of the interactions required by the given exploration ID.
(Is it ok?, as exploration ID is the arg for this function)

Reply:
Yes, that's fine.